src/public/js/main.js

```diff
@@ -15,16 +17,33 @@
       }
 
       var msg = $('#post').val();
-      var encoded = RS.encode(msg);
+      var i = 1;
+      var last = 0;
+      var encoded = new Array(0);
+      for (; i <= msg.length; i++) {
+        if (i % k == 0) {
+          Array.prototype.push.apply(encoded, RS.encode(msg.slice(i - k, i)));
+          last = i;
+        }
+      }
+      Array.prototype.push.apply(encoded, RS.encode(msg.slice(last, i)));
       socket.emit(CONSTANTS.EVENT_TYPES.MESSAGE, encoded);
       $('#post').val('');
       return false;
     });
 
     // upon broadcasted message from server, decode message and add to screen
     socket.on(CONSTANTS.EVENT_TYPES.MESSAGE, function(encoded) {
-      var decoded = RS.decode(encoded);
-      $('#messages').append($('<p>').text('> ' + decoded));
+      var i = 1;
+      var decoded = new Array(0);
+      for (; i <= encoded.length; i++) {
```

Comment:
This one is not necessary :) I coppied the loop from the encode which does need the i to be declared outside the loop scope, but on decode there is no need so I shall fix that 👍